docs/proposals/2018-08-07-Task_Prompted_Media.md

```diff
@@ -0,0 +1,35 @@
+# TASK-PROMPTED MEDIA
+
+# Problem
+
+Most compelling content includes more than one medium - text, photography, graphics, etc. 
+
+In order to coordinate the production of this content, editors can create tasks to request colloboration from team's with different skills.
+
+If the team decides to take on the work, they'll accept the task and dole out the assignment. 
+
+At that point, the media is either created (the sidebar is written, the graphic is created) or sourced (an appropriate asset is found on the wires or in the archive). 
+
+Now that the media exists, the editor who originally commisionshed the work needs a way to easily locate it. This doesn't exist today within Arc. 
+
+Allowing content creators to tag media with task IDs will make it easy to find work that is produced as the result of a task.
+
+# Proposal
+
+A new field, associated_tasks, will allow an array of Task IDs (strings) to be associated with any content inside of Arc. This new property will belong to `trait_taxonomy`
+
+
+    "associated_tasks": {
+      "description": "A list of task IDs that this content was created or curated to satisfy.",
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+
+# Concerns
+
+- Do we need to limit the number of task IDs that can be associated with a story? It is conceivable that a single photo, for example, is used to satsify many different tasks.
```

Comment:
There isn't a spec for the ID. I think providing a character and array limit is a good idea.

Reply:
Also, I can see benefit with limiting the string IDs themselves, but what is the benefit of limiting the array length?  @awong7 makes a good point about an iconic image which might continue to be reused for various tasks over time. If that seems like a valid use case, then what happens when this limit is reached?

Reply:
If the former, we probably ought to rename the field here to be clearer. I think that's much better than using an overly general name and only specifying the restriction in the doc. To me it just doesn't seem reasonable to assume that we'll never have another system trying to do some similar integration with content objects.

Reply:
While we're on the subject of task ID names, I always thought it might make sense to use a combination of words as the ID instead of alphanumeric characters if journalists are going to have to be hand-copying or shouting and quickly relaying the ID name to each other. For example, the task ID will be relayed from a photo editor to freelance photographers to enter in external photo software they use and a combination of words is easier to remember. 

@calexicoz used a random repo name generator for a personalization team repo, and I will never forget "creepy-rutabaga" even though that was 2+ years ago. https://appear.in/ also uses this concept for its video chat links (ex: appear.in/awesome-cat) since you're often shouting or quickly relaying video chat links to your coworker.

Reply:
`if journalists are going to have to be hand-copying or shouting and quickly relaying the ID name to each other` surfacing the task id is almost entirely a stop gap measure until we can make the cross app experience fully seamless, otherwise I think this is an interesting idea.

Reply:
I'm with @chadmoone. If this is purely for WebSked, then the property name needs to make that clear. If there is a potential for other uses, then we need namespacing. I would also argue along @willvanwazer that if this is to be a generic field, then we should allow objects, with probably a minimum required field of `_id`. Otherwise, if it's for WebSked, then the field needs to be called "WebSked something".

Also, namespacing is not the end of the world. WebSked can continue to use mongo IDs, and we could just use `websked:ID` in the field. It's up to the applications to break at the `:` to get the actual ID when reading/writing.

Reply:
And `creepy-rutabaga`, I remember you too... We even got blacklisted by some anti-virus or something because of it.. Fun times. @awong7 ;)

Reply:
@calexicoz @chadmoone We have never asked contributors to insert the name of their application in the property in the past -- imposing this requirement now would be making a change in the de facto style guide of ANS.    There are many, many properties in ANS that can only be meaningfully editied/viewed in Arc within a single application, but we've always taken the view that the property name should describe what the property does, not who owns/manages it.  Generally the application-specific context is included in the JSON schema description, prefaced with something like "In Arc, this is used by Content API when doing xxxx" or something along those lines.

Perhaps that implicit style -- considering "ANS" to be a standard to which Arc simply happens to adhere to -- has outlived its usefulness, but I don't think the best way to change that is suddenly, in this property only, force the field name to include "websked" in it.

Reply:
Thanks for the context @gengel. That makes more sense to me.

Reply:
Thanks for the context, @gengel. That's very helpful to know.

Reply:
Yeah, I know that namespacing the IDs wouldn't be a huge lift but I'm pretty skeptical that we're talking about something (managing WebSked-style tasks while expecting it to integrate nicely with the rest of Arc) that clients are going to develop as bolt-on custom software.